src/main/scala/TraceIO.scala

```diff
@@ -18,28 +18,33 @@ import freechips.rocketchip.diplomacy.{BundleBridgeSource, BundleBroadcast, Bund
 // used to connect the TracerV or Dromajo bridges (in FireSim and normal sim)
 //***************************************************************************
 
-case class TracedInstructionWidths(iaddr: Int, insn: Int, wdata: Int, cause: Int, tval: Int)
+case class TracedInstructionWidths(iaddr: Int, insn: Int, wdata: Option[Int], cause: Int, tval: Int)
 
 object TracedInstructionWidths {
-  def apply(tI: ExtendedTracedInstruction): TracedInstructionWidths =
-    TracedInstructionWidths(tI.iaddr.getWidth, tI.insn.getWidth, tI.wdata.getWidth, tI.cause.getWidth, tI.tval.getWidth)
+  def apply(tI: ExtendedTracedInstruction): TracedInstructionWidths = {
+    val wdataWidth = tI.wdata match {
```

Comment:
I'm reviewing this from top to bottom but again the more succinct scala-y way to write this `tI.wdata map _.getWidth`.